tap-agent/src/tap/escrow_adapter.rs

```diff
@@ -74,16 +89,37 @@ impl EscrowAdapterTrait for EscrowAdapter {
             .sender_pending_fees
             .read()
             .await
-            .get(&sender)
+            .get(sender)
             .copied()
             .unwrap_or(0);
         Ok(balance - fees)
     }
 
     async fn subtract_escrow(&self, sender: Address, value: u128) -> Result<(), AdapterError> {
+        let escrow_accounts =
+            self.escrow_accounts
+                .value()
+                .await
+                .map_err(|e| AdapterError::AdapterError {
+                    error: format!("Could not get escrow accounts from eventual: {:?}.", e),
+                })?;
+
         let current_available_escrow = self.get_available_escrow(sender).await?;
+
+        let sender =
+            escrow_accounts
+                .signers_to_senders
+                .get(&sender)
+                .ok_or(AdapterError::AdapterError {
+                    error: format!(
+                        "Sender {} not found for receipt signer, could not get available escrow.",
+                        sender
+                    )
+                    .to_string(),
+                })?;
```

Comment:
I think I'd like to have a `get_sender_for_signer(signer: Address) -> Result<Address>` because it hides the data structure used and also we don't need to do `ok_or()` all over the place.